runtime/vm/jvminit.c

```diff
@@ -1878,12 +1878,13 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 					vm->vmRuntimeStateListener.idleTuningFlags &= ~(UDATA)J9_IDLE_TUNING_GC_ON_IDLE;
 				}
 				/*
-				 * CompactOnIdle is enabled only if:
+				 * CompactOnIdle is enabled only if -XX:+IdleTuningGcOnIdle is set and:
 				 * 1. -XX:+IdleTuningCompactOnIdle is set, or
 				 * 2. running in container
 				 */
-				if ((argIndexCompactOnIdleEnable > argIndexCompactOnIdleDisable)
-				|| ((-1 == argIndexCompactOnIdleDisable) && inContainer)
+				if ((TRUE == enableGcOnIdle)
```

Comment:
This test already occurs a few lines up.  Can this block of code be moved into the existing if statement - line 1875?

Reply:
Moved the code block to inside the if condition.